### ISSUE TITLE
guzzlehttp/guzzle:^7.2 install issue with this

### ISSUE THREAD

**Issue by emtiazzahid** [GrahamCampbell/Laravel-GitHub#136]
**PHP version**: 7.4.19 
**Description**
Tried to install with fresh new laravel ([8.65](https://github.com/laravel/framework/releases/tag/v8.65.0)) project
when tried to run that command as instructed on documentation
`composer require "graham-campbell/github:^10.3" "guzzlehttp/guzzle:^7.2" "http-interop/http-factory-guzzle:^1.0" `

Log: 
` Problem 1
    - Root composer.json requires guzzlehttp/guzzle 7.2 -> satisfiable by guzzlehttp/guzzle[7.2.0].
    - guzzlehttp/guzzle 7.2.0 requires guzzlehttp/psr7 ^1.7 -> found guzzlehttp/psr7[1.7.0, ..., 1.x-dev] but the package is fixed to 2.1.0 (lock file version) by a partial update and that version does not match. Make sure you list 
it as an argument for the update command.
  Problem 2
    - Root composer.json requires http-interop/http-factory-guzzle 1.0 -> satisfiable by http-interop/http-factory-guzzle[1.0.0].
    - http-interop/http-factory-guzzle 1.0.0 requires guzzlehttp/psr7 ^1.4.2 -> found guzzlehttp/psr7[1.4.2, ..., 1.x-dev] but the package is fixed to 2.1.0 (lock file version) by a partial update and that version does not match. Ma
ke sure you list it as an argument for the update command.
  Problem 3
    - guzzlehttp/guzzle 7.2.0 requires guzzlehttp/psr7 ^1.7 -> found guzzlehttp/psr7[1.7.0, ..., 1.x-dev] but the package is fixed to 2.1.0 (lock file version) by a partial update and that version does not match. Make sure you list 
it as an argument for the update command.
    - graham-campbell/github v10.3.0 requires knplabs/github-api 3.3.* -> satisfiable by knplabs/github-api[v3.3.0].
    - knplabs/github-api v3.3.0 requires psr/http-client-implementation ^1.0 -> satisfiable by guzzlehttp/guzzle[7.2.0].
    - Root composer.json requires graham-campbell/github 10.3 -> satisfiable by graham-campbell/github[v10.3.0].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
`

**How to reproduce**
Install [laravel v8.65](https://github.com/laravel/framework/releases/tag/v8.65.0)
Run `composer require "graham-campbell/github:^10.3" "guzzlehttp/guzzle:^7.2" "http-interop/http-factory-guzzle:^1.0"`

**Possible Solution**
with install updated version of guzzle. need to update readme
`composer require "graham-campbell/github:^10.3" "guzzlehttp/guzzle:^7.4" "http-interop/http-factory-guzzle:^1.1"`

**Comment by GrahamCampbell** [comment]
Thanks for the report. This looks like a composer bug to me. Those version constraints are exactly the same, apart from the minimum versions bumped.

**Comment by GrahamCampbell** [comment]
I think the underlying issue is that the http factory interop 1.0.x line doesn't support the newer guzzle psr7. If you ran `composer update` first, the original command would probably work. I'll look into updating the README.

**Comment by ahinkle** [comment]
Hey @GrahamCampbell ,

I'm seeing the same on a fresh Laravel application. I ran a fresh composer update.  Any ideas? It's getting tripped up on knplabs/github-api 3.3:

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - graham-campbell/github[v10.3.0, ..., 10.3.x-dev] require knplabs/github-api 3.3.* -> satisfiable by knplabs/github-api[v3.3.0].
    - knplabs/github-api v3.3.0 requires symfony/deprecation-contracts ^2.2 -> found symfony/deprecation-contracts[v2.2.0, ..., 2.5.x-dev] but the package is fixed to v3.0.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires graham-campbell/github ^10.3 -> satisfiable by graham-campbell/github[v10.3.0, v10.3.1, 10.3.x-dev].
  ```

Composer.json:

`composer require "graham-campbell/github:^10.3" "guzzlehttp/guzzle:^7.2" "http-interop/http-factory-guzzle:^1.0"`


```json
    "require": {
        "php": "^7.3|^8.0",
        "fruitcake/laravel-cors": "^2.0",
        "guzzlehttp/guzzle": "^7.2", // <-- I modified this from 7.0.1
        "laravel/framework": "^8.65",
        "laravel/sanctum": "^2.11",
        "laravel/tinker": "^2.5"
    },
    "require-dev": {
        "facade/ignition": "^2.5",
        "fakerphp/faker": "^1.9.1",
        "laravel/sail": "^1.0.1",
        "mockery/mockery": "^1.4.4",
        "nunomaduro/collision": "^5.10",
        "phpunit/phpunit": "^9.5.10"
    },
```

**Comment by stephenstack** [comment]
same issue as @ahinkle

**Comment by stephenstack** [comment]
knplabs/github-api composer dependencies are outdate to older versions of psr/cache and symfony/deprecation-contracts. they need to be updated on that project

**Comment by emtiazzahid** [comment]
@stephenstack Worked for me

`composer require "graham-campbell/github:^10.3" "guzzlehttp/guzzle:^7.4" "http-interop/http-factory-guzzle:^1.1"`

**Comment by stephenstack** [comment]
Brand new laravel install on php8. See output below.

[root@homedev testapp]# php -v
PHP 8.0.14 (cli) (built: Dec 16 2021 03:01:07) ( NTS gcc x86_64 )
Copyright (c) The PHP Group
Zend Engine v4.0.14, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.14, Copyright (c), by Zend Technologies
[root@homedev testapp]# php artisan --version
Laravel Framework 8.79.0
[root@homedev testapp]# composer require "graham-campbell/github:^10.3" "guzzlehttp/guzzle:^7.4" "http-interop/http-factory-guzzle:^1.1"
./composer.json has been updated
Running composer update graham-campbell/github guzzlehttp/guzzle http-interop/http-factory-guzzle
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - knplabs/github-api v3.3.0 requires symfony/deprecation-contracts ^2.2 -> found symfony/deprecation-contracts[v2.2.0, ..., 2.5.x-dev] but the package is fixed to v3.0.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - graham-campbell/github[v10.3.0, ..., 10.3.x-dev] require knplabs/github-api 3.3.* -> satisfiable by knplabs/github-api[v3.3.0].
    - Root composer.json requires graham-campbell/github ^10.3 -> satisfiable by graham-campbell/github[v10.3.0, v10.3.1, 10.3.x-dev].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
